### DISCUSSION TITLE
"Cross-site POST form submissions are forbbiden" when deploying on DigitalOcean Apps

### DISCUSSION THREAD

**Discussion by paulchoix** [sveltejs/kit#9020] [Q&A]
I've been using `@sveltejs/adapter-node` on DigitalOcean Apps and managed to get it up and running as long as I set `checkOrigin` to false, which I understand is not ideal. I've seen a few posts about setting the `ORIGIN` environment variable but it doesn't seem to work.

On DigitalOcean Apps, my component settings are as follows:

- Build Command: `npm run build`
- Run Command: `npm run preview`
- HTTP Port: 4173

The Apps interface allows you to set environment variables, but the [posts I've seen online](https://stackoverflow.com/a/73821896/12421409) mention prefixing `npm run build` with `ORIGIN=http://localhost:3000` (such as `ORIGIN=http://localhost:3000 npm run build`). I'm not sure if these two methods are identical. Meanwhile, the [adapter-node docs](https://kit.svelte.dev/docs/adapter-node) mention `ORIGIN=https://my.site node build`. To add to the confusion, the port is 4173 in my setup, but both the docs and the posts I've seen speak of port 3000 (if I'm not mistaken, this is the default port used by Node, while Vite's `preview` uses 4173).

I've tried every combination based on the previous information:

- Setting an `ORIGIN` environment variable (in the Apps interface) as `http://localhost:3000`, `http://localhost:4173` and `https://my.site`
- Prefixing the build command with `ORIGIN=http://localhost:3000`, `ORIGIN=http://localhost:4173` and `ORIGIN=https://my.site`

None of these work. I'm starting to wonder if DigitalOcean's deployment process is overriding something, or if my configuration is faulty in some other way. There's also a set of "CORS Policy" settings in the Apps interface, but I don't think they are relevant (I've left them as "Not Configured"). In my browser network tab, the origin header is always `https://my.site` no matter what setup I use. It also may be worth noting that I'm currently testing using the `ondigitalocean.app` domain provided by DigitalOcean. Any help?

**Comment by wvhulle** [comment]
You don't have to prefix the build command with origin. It's the node build/index.js command.

**Reply by paulchoix** [reply]
Is that referring to the Run Command in DigitalOcean Apps? I tried prefixing `npm run preview` with all three origins mentioned above, but it didn't work.
